pkg/operator/targetconfigcontroller/targetconfigcontroller.go

```diff
@@ -277,6 +278,16 @@ func managePod(client coreclientv1.ConfigMapsGetter, recorder events.Recorder, o
 	}
 	required.Spec.Containers[0].Args = append(required.Spec.Containers[0].Args, fmt.Sprintf("-v=%d", v))
 
+	var observedConfig map[string]interface{}
+	if err := yaml.Unmarshal(operatorConfig.Spec.ObservedConfig.Raw, &observedConfig); err != nil {
+		return nil, false, fmt.Errorf("failed to unmarshal the observedConfig: %v", err)
+	}
+	proxyConfig, _, err := unstructured.NestedStringMap(observedConfig, "proxy")
```

Comment:
would choose a more unique field. Not that we add a proxy field to the official type and this break down.

targetconfigcontroller.proxy ?

Reply:
Went with `targetconfigcontroller.proxy` then